File: plugin/stripcomments/fixtures/commented.py
"doc string here"
# single line comment
somevar = '''this is some correct value''' # accompanied with a comment
print(somevar) # this is also a comment

'''
Docstring outside func
'''
def func(arg: str):
    '''
    Docstring inside func
    '''
    print("func-", arg) ###### comment here
    def nested_func():
        """
        Docstring inside nested func
        """
        if 4<9:
            if 5<3:
                print("deeper_nested_func")
                # single line comment
            else:
                print("deeper_nested_func")
                while(1):
                    print("deepest_nested_func") ###### comment here
                    if 5>3:
                        break
                match 'a':
                    case 'a':
                        if 3<10:
                            print("a_if")
                            '''further comment'''
                        print("a")
                    case 'b':
                        match "zs":
                            # single line comment
                            case "zs":
                                print("zs")
                            case _:
                                print("default")
                        print("b") # some comments
                    case _:
                        print("default")###
            print("deep_nested_func")
        print("nested_func")#
        pass
    nested_func()
    pass

# print("Commented out code")
print("log1")#
func("don't mistake this string for comment - # this is valid code")
func("""some str""")##

class MyClass:
    """
    Docstring inside class
    """
    def __init__(self, arg: str):
        '''Docstring inside class function'''
        print("MyClass-", arg)
        pass
    print("log7")
    pass

"""
All these are valid expressions
"""
g = MyClass("gggg")
g = MyClass('gggg')
g = MyClass('''gggg''')
g = MyClass("""gggg""")

File: plugin/stripcomments/fixtures/commented.js
import os from 'os' // This is a single line comment
const express = require('express'); ////// This is also single line comment

os.homedir(); // This is also a single line comment

const x = 1; /* This is a multi line comment */

/*
Multi line comment docstring
console.log('This is a commented out code');
console.log('This is a commented out code');
*/
// Function with commented-out parts
function sum(a, b) {
// const temp = a + b;
  return a + b; /* This is the actual sum operation */
}
sum(5, 10);

// console.log('This is a commented out code');
console.log('This is valid code');
// console.log('This is a commented out code');

/**
 * This is a docstring for Class Test
 * This is a docstring for Class Test
 * This is a docstring for Class Test
 */
class Test {
  constructor() {
    console.log('Test class created');
  }
  /**
   * arg1 - argument 1 of type string
   * arg2 - argument 2 of type string
   */
  test(arg1, arg2) {
    console.log(arg1); /**/
    console.log(arg2); /// 
  }
}

switch ("red") {
  case 'blue':
    console.log('Color is blue'); // Comment inside case
    break;
  case 'red':
    /* This is the red color block */
    console.log('Color is red'); // Comment inside case for red
    break;
  default:
    console.log('Unknown color'); // Default case
    // console.log('Color is unknown');
}


